src/utils/RealtimeFeedUtils.js

```diff
@@ -172,8 +87,33 @@ function getDelayInformation(stopID: String, tripID: String, rtf: Object): ?Obje
   };
 }
 
+function getVehicleInformation(
+  routeID: String,
+  tripID: String,
+  vehicles: Object,
+): ?Array<Object> {
+  // vehicles param ensures the vehicle tracking information doesn't update in
+  // the middle of execution
+  if (!routeID
+    || !tripID
+    || !vehicles
+    || vehicles === {}) {
+    LogUtils.log({
+      category: 'getVehicleInformation NULL',
+      routeID,
+      tripID,
+    });
+    return null;
+  }
+  return Object.values(vehicles).filter(
+    v => v.routeID === routeID && v.tripID === tripID,
+  )[0];
```

Comment:
this would error if the return of `filter` is an empty array right? we should prob check the length before returning `[0]`

Reply:
u right.. i shall defer to your way :) thx